File: src/main/java/com/sica/backend/repository/DivisionRepository.java
package com.sica.backend.repository;

import com.sica.backend.entity.Division;
import com.sica.backend.entity.Level;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface DivisionRepository extends JpaRepository<Division, String> {
    List<Division> findByLevel(Level level);
    List<Division> findByActiveTrue();
    List<Division> findByLevelAndActiveTrue(Level level);
}

File: src/main/kotlin/com/sica/backend/repository/DivisionRepository.kt
package com.sica.backend.repository

import com.sica.backend.entity.Division
import com.sica.backend.entity.Level
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface DivisionRepository : JpaRepository<Division, String> {
    fun findByLevel(level: Level): List<Division>
    fun findByActiveTrue(): List<Division>
    fun findByLevelAndActiveTrue(level: Level): List<Division>
}
